src/main/java/com/gmail/goosius/siegewar/utils/SiegeWarDominationAwardsUtil.java

```diff
@@ -299,20 +304,74 @@ public static List<Nation> getNationsListSortedNormally() {
+        return item.getItemMeta() != null
+                && item.getItemMeta().getPersistentDataContainer().has(EXPIRATION_TIME_KEY, EXPIRATION_TIME_KEY_TYPE);
    }
+
+    /**
+    * Determine is a given item is an expired artefact
+    * @param item the item
+    *
+    * @return true if the item is an expired artefact
+    */
+    public static boolean isExpiredArtefact(ItemStack item) {
+        PersistentDataContainer persistentDataContainer;
+        Long expiryTime;
+        if(item.getItemMeta() != null) {
+            persistentDataContainer = item.getItemMeta().getPersistentDataContainer();
+            expiryTime = persistentDataContainer.get(EXPIRATION_TIME_KEY, EXPIRATION_TIME_KEY_TYPE);
+            return expiryTime != null && System.currentTimeMillis() > expiryTime;
+        } else {
+            return false;
+        }
+   }
+
+    /**
+     * This method scans a certain number of the online players
+     * 
+     * If any of those online players are carrying expired artefacts:
+     * 1. All their artefacts are deleted.
+     * 2. All their artefacts explode.
+     */
+    public static void evaluateArtefactExpiries() {
+        if(!SiegeWarSettings.isDominationAwardsGlobalEnabled())
+            return;
+        //Decide which players to scan
+        List<Player> playersToScan = new ArrayList<>();
+        double normalizedScanChance;
+        for(Player player: Bukkit.getOnlinePlayers()) {
+            normalizedScanChance = SiegeWarSettings.getDominationAwardsArtefactExpiryPercentageChancePerShortTick() / 100;
+            if(Math.random() < normalizedScanChance) {
+                playersToScan.add(player);
+            }
+        }
+        if(playersToScan.size() == 0)
+            return;
+        //Scan the players
+        int numExpiredArtefacts;
+        int explosionPower;
+        for(Player player: playersToScan) {
+            //Delete Artefacts
+            numExpiredArtefacts = 0;
+            for(ItemStack item: player.getInventory().getContents()) {
+                if(item != null && isExpiredArtefact(item)) {
+                    item.setAmount(0);
+                    numExpiredArtefacts++;
+                }
+            }
+            //Create explosion
+            if(numExpiredArtefacts > 0 && SiegeWarSettings.getDominationAwardsArtefactExpiryExplosionsEnabled()) {
+                explosionPower = SiegeWarSettings.getDominationAwardsArtefactExpiryExplosionsBasePower()
+                                    + (SiegeWarSettings.getDominationAwardsArtefactExpiryExplosionsExtraPowerPerExpiredArtefact() * numExpiredArtefacts);
+                int finalExplosionPower = Math.min(explosionPower, SiegeWarSettings.getDominationAwardsArtefactExpiryExplosionsMaxPower());                
+                SiegeWar.getSiegeWar().getServer().getScheduler().runTask(SiegeWar.getSiegeWar(), ()->  player.getWorld().createExplosion(player.getLocation(), finalExplosionPower, true));
```

Comment:
```suggestion
                Bukkit.getScheduler().runTask(SiegeWar.getSiegeWar(), ()-> player.getWorld().createExplosion(player.getEyeLocation(), finalExplosionPower, true));
```

Simplify things and make the explosion right in their face. (Face part optional.)

Reply:
Haha. Ok sounds good! :-D